packages/kbn-observable/src/factories/from_iterable.ts

```diff
@@ -0,0 +1,26 @@
+import { Observable } from '../observable';
+
+function isIterable<T>(obj: any): obj is Iterable<T> {
+  // checks for null and undefined
+  if (obj == null) {
+    return false;
+  }
+  return typeof obj[Symbol.iterator] === 'function';
+}
+
+/**
+ * Create an observable from an iterable
+ */
+export function $fromIterable<T>(x: Iterable<T>): Observable<T> {
+  if (!isIterable(x)) {
+    throw new TypeError(`${x} is not iterable`);
+  }
+
+  return new Observable<T>(observer => {
+    for (let item of x) {
+      observer.next(item);
+    }
```

Comment:
We should be checking for `observer.closed` after calling `observer.next()` in case x is infinitely iterable.

Reply:
You might mean something else, but we already cover `.closed` in https://github.com/kjbekkelund/kibana/blob/4ad05989c8f0b61790f4ff4e6027d301c1efa6e5/packages/kbn-observable/src/observable.ts#L144. Or are you thinking of something else? Got a suggestion for a failing test?

Reply:
Hm, thinking more about this: won't an infinite observable just "lock up" here and eventually oom?

Reply:
Here's a failing test for you:

```ts
test('support infinitely iterable values', async () => {
  function* counter() {
    let i = 0;
    while (true) {
      yield i++
    }
  }

  const infiniteCounter = counter()
  const values: number[] = [];
  const error = jest.fn();
  const complete = jest.fn();

  $fromIterable(infiniteCounter).pipe(take(4)).subscribe({
    next(x) {
      values.push(x);
    },
    error,
    complete,
  });

  expect(values).toEqual([0, 1, 2, 3]);
  expect(error).not.toHaveBeenCalled();
  expect(complete).toHaveBeenCalledTimes(1);
});
```

The reason it doesn't lock up the process is because it's limited by a downstream operator, `take()` unsubscribes when it's done which is why `$fromIterable()` should check for `observer.closed` after delivering each notification with `observer.next()`.